ui/core/components/suggest_reforges_action.tsx

```diff
@@ -2021,4 +2091,49 @@ export class ReforgeOptimizer {
 			delay: 10000,
 		});
 	}
+
+	fromProto(eventID: EventID, proto: ReforgeSettings) {
+		TypedEvent.freezeAllAndDo(() => {
+			this.setUseCustomEPValues(eventID, proto.useCustomEpValues);
+			this.setStatCaps(eventID, Stats.fromProto(proto.statCaps));
+			this.setUseSoftCapBreakpoints(eventID, proto.useSoftCapBreakpoints);
+			this.setIncludeTimeout(eventID, proto.includeTimeout);
+			this.setIncludeGems(eventID, proto.includeGems);
+			this.setIncludeEOTBPGemSocket(eventID, proto.includeEotbGemSocket);
+			this.setFreezeItemSlots(eventID, proto.freezeItemSlots);
+			this.setFrozenItemSlots(eventID, proto.frozenItemSlots);
+			this.setBreakpointLimits(eventID, Stats.fromProto(proto.breakpointLimits));
+			if (proto.relativeStatCapStat) {
+				this.setRelativeStatCap(eventID, UnitStat.fromProto(proto.relativeStatCapStat).getStat());
+			}
+		});
+	}
+	toProto(): ReforgeSettings {
+		return ReforgeSettings.create({
+			useCustomEpValues: this.useCustomEPValues,
+			useSoftCapBreakpoints: this.useSoftCapBreakpoints,
+			includeTimeout: this.includeTimeout,
+			includeGems: this.includeGems,
+			includeEotbGemSocket: this.includeEOTBPGemSocket,
+			freezeItemSlots: this.freezeItemSlots,
+			frozenItemSlots: [...this.frozenItemSlots],
+			breakpointLimits: this.breakpointLimits.toProto(),
+			relativeStatCapStat: this.relativeStatCap?.forcedHighestStat.toProto(),
```

Comment:
Also make sure to update the `toProto()` method on `UnitStat` to write the current API version, in order to future proof it for any required migrations if the schema ever changes.